src/nodes/FunctionDefinition.js

```diff
@@ -6,10 +6,21 @@ const {
 
 const printList = require('./print-list');
 
-const functionName = node => {
+const functionName = (node, options) => {
   if (node.isConstructor && !node.name) return 'constructor';
   if (node.name) return `function ${node.name}`;
-  return 'function';
+  if (node.isReceiveEther) return 'receive';
+  // The parser doesn't give us any information about the keyword used for the
+  // fallback.
+  // Luckily, `function` and `fallback` have 8 characters so using the
+  // originalText is the next best option.
+  //
+  // An neat idea would be to rely on the pragma and enforce it but for the
+  // moment this will do.
+  return options.originalText.slice(
```

Comment:
should we enforce here that we should return only `function` and `fallback` and not, for instance, `facepalm` (yes it's 8 letters)?

Reply:
Can do that but will mess with coverage though as there’s no chance for us to trigger the parser to have `facepalm` as a function definition

Reply:
ok I updated it to be a bit more explicit

Reply:
hahaha, I've pushed a commit, what do you think? something in between?

Reply:
yup, this way we'll also know if solidity inserts a new keyword as `undefined` doesn't silently print in `concat`

Reply:
I go ahead and merge then ok?